internal/server/proto/server.proto

```diff
@@ -901,12 +901,12 @@ message Generation {
 
 // A platform resource that an operation (release/deployment) has created, depends on, or manages.
 message DeclaredResource {
-  // The id of the declared resource. Automatically generated
-  string id = 6;
-
   // platform-specific name of the resource type. i.e. instance, pod, auto-scaling group, etc
   string name = 1;
 
+  // platform-specific type of the resource type. i.e. instance, pod, auto-scaling group, etc
+  string type = 6;
```

Comment:
nit: I feel like this should be at the end of the message definition, so the field numbers stay in order. Do we have a consensus or pattern placing new fields in the proto definition sequentially by field number?

Reply:
It looks like we usually bias toward making them read in a logical order, rather than maintain numerical order (ex: the Deployment message)
https://github.com/hashicorp/waypoint/blob/3bf1b26fd89650aee52c27db756a1b8c9dfa4d45/internal/server/proto/server.proto#L2547-L2558